### PR TITLE
Fix `rcond` arg of `linalg.lstsq`

### PR DESCRIPTION
**Pull request by toslunar**
- The default is `rcond='warn'`, which fallbacks to `rcond=-1`.
- Negative `rcond` should be treated as the machine precision.
- NumPy's future default `rcond=None` is max(m, n) times machine precision.

### CODE REVIEW

**Comment by takagi** [comment]
pfnCI, test this please.

File: cupy/linalg/_solve.py [hunk 1]
```diff
@@ -223,6 +225,18 @@ def lstsq(a, b, rcond=1e-15):
...
+        warnings.warn(
+            '`rcond` parameter will change to the default of '
+            'machine precision times ``max(M, N)`` where M and N '
+            'are the input matrix dimensions.\n'
+            'To use the future default and silence this warning '
+            'we advise to pass `rcond=None`.\n'
+            'Using the old CuPy default: `rcond=1e-15`, which is '
+            'neither the same as the old NumPy default (`rcond=-1`).',
+            FutureWarning)
+        rcond = 1e-15
```
Comment (toslunar) [inline cupy/linalg/_solve.py:238]: The different default value is considered as a bug.
```suggestion
        rcond = -1
```

Then, the function should also implement the following spec of `*gelss`:

> If RCOND < 0, machine precision is used instead.

Comment (toslunar) [reply cupy/linalg/_solve.py:238]: It seems a reference implementation uses `if rcond >= 0 or rcond <= 1`.  I found the following code in `dlalsd`.
```fortran
       IF( ( rcond.LE.zero ) .OR. ( rcond.GE.one ) ) THEN
          rcnd = eps
       ELSE
          rcnd = rcond
       END IF
```

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
LGTM! Waiting for CI pass.

**Pull request merged**